pluginsdk/support/bundlepublisherutil/bputil.go

```diff
@@ -0,0 +1,257 @@
+// Package bundlepublisherutil provides helper functions for plugins
+// implementing the BundlePublisher interface.
+// BundlePublisher plugins should use this package as a way to have a
+// standarized name for bundle formats in their configuration, and avoid the
+// re-implementation of bundle parsing logic of formats supported in this
+// package.
+package bundlepublisherutil
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/spiffebundle"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/types"
+	"gopkg.in/square/go-jose.v2"
+)
+
+const (
+	BundleFormatUnset BundleFormat = iota
+	SPIFFE
+	PEM
+	JWKS
+)
+
+// Bundle represents a bundle that can be formatted in different formats.
+type Bundle struct {
```

Comment:
To be aligned with other types that do similar things, how would you feel about the name `Formatter`?